### ISSUE TITLE
Incorrect test - reference to algorithm and not param

### ISSUE THREAD

**Issue by DinoChiesa** [w3c-ccg/http-signatures-test-suite#24]
https://github.com/w3c-ccg/http-signatures-test-suite/blob/d202f702389f923e1586d733af55339c95c8fded/test/latest/10-canonicalize.js#L265

According to the comments, this test purports to test the presence of (created) and (expires) header parameters when the algorithm is one of rsa-* hmac-* or ecdsa-* . But it appears to me based on code inspection that this line ought to use

   generatorOptions.args.headers = [`(${param})`];

and not

  generatorOptions.args.headers = [`(${algorithm})`];

Also, in line 269 I think it ought to be

                  `${param}-${algorithm}`, generatorOptions);

and not
                  `created-${algorithm}`, generatorOptions);

There may be other problems here. It probably warrants taking a closer look.

**Comment by aljones15** [comment]
You are correct about like 269 (I don't how I missed that) as right now it only tests `created-${algorithm}`:

Here is an example:
https://github.com/w3c-ccg/http-signatures-test-suite/blob/d202f702389f923e1586d733af55339c95c8fded/test/latest/input/created-ecdsa.httpMessage

Also seem to be missing httpMessages for `expires-${algorithm}` so my apologies there. I will inspect the resulting httpMessages and see if I can correct this. Also be aware that 2 tests product false negatives in this current batch. I'm also sorry about that. I will look into this, but not sure when. Feel free to assign this issue to me as I would really like to review these tests and correct a lot of the little mistakes in them.

**Comment by aljones15** [comment]
@DinoChiesa just so you know `expires` needs to be set dynamically so that is a unix time stamp greater than the current time (I think 10 minutes is pretty safe). That appears to be why `expires` was skipped (we have to generate an httpMessage dynamically in this case). So fixing this might require some time as it requires rewriting the test suite so we can format and output httpMessages in a way that works across multiple platforms. I believe I tried doing this when writing the tests originally, but ran into formatting issues. Maybe I can succeed this time. As for passing the algorithm name in `--headers` that might be incorrect. I checked and in our implementation if there are no headers specified to use to form the signature then `(created)` is added automatically. I believe that is in the spec. I'll need to check the spec and see how poorly this test confirms to it (probably pretty bad) and what the the `headers` parameter should be in this case.
